presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java

```diff
@@ -999,39 +999,71 @@ public void testCountColumn()
     }
 
     @Test
-    public void testWildcard()
+    public void testSelectAllFromTable()
     {
         assertQuery("SELECT * FROM orders");
-    }
 
-    @Test
-    public void testMultipleWildcards()
-    {
+        // multiple wildcards
         assertQuery("SELECT *, 123, * FROM orders");
-    }
 
-    @Test
-    public void testMixedWildcards()
-    {
+        // mixed wildcards
         assertQuery("SELECT *, orders.*, orderkey FROM orders");
-    }
 
-    @Test
-    public void testQualifiedWildcardFromAlias()
-    {
+        // qualified wildcard from alias
         assertQuery("SELECT T.* FROM orders T");
-    }
 
-    @Test
-    public void testQualifiedWildcardFromInlineView()
-    {
+        // qualified name from alias, column aliases
+        assertQuery("SELECT T.* FROM (VALUES (1, 2, 3)) T (a, b, c)", "SELECT 1 a, 2 b, 3 c");
+
+        //qualified wildcard from inline view
         assertQuery("SELECT T.* FROM (SELECT orderkey + custkey FROM orders) T");
+
+        // qualified wildcard
+        assertQuery("SELECT orders.* FROM orders");
     }
 
     @Test
-    public void testQualifiedWildcard()
+    public void testSelectAllFromRow()
     {
-        assertQuery("SELECT orders.* FROM orders");
+        // named row fields and aliases
+        assertQuery("SELECT CAST((ROW (1, 'a')) AS ROW (f1 bigint, f2 varchar(1))).* AS (alias1, alias2)", "SELECT 1 alias1, 'a' alias2");
+
+        // unnamed row fields and aliases
+        assertQuery("SELECT (ROW (1, 'a')).* AS (alias1, alias2)", "SELECT 1 alias1, 'a' alias2");
```

Comment:
If I'm not mistaken, `assertQuery` doesn't validate that the output columns are named the same. So, `SELECT 1 alias1, 'a' alias2` is not testing what you think. I'm not sure we even have a way in the current framework to do so.

Maybe replace it with something like this:

```sql
SELECT alias1, alias2 
FROM (
    SELECT (ROW (1, 'a')).* AS (alias1, alias2)
) t
```

Then we can be sure that the aliasing is working fine.